From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fredrik=20S=C3=B6derquist?= <fs@opera.com>
Date: Tue, 20 Oct 2026 10:09:46 -0700
Subject: [M156] Make timed elements inserted into <use> shadow trees inert
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Original change's description:
> Make timed elements inserted into <use> shadow trees inert
>
> If timed elements somehow make it into a <use> shadow tree, they should
> not be scheduled on the timeline. Treat them the same as if they were
> not in the document at all.
>
> Fixed: 574791952
> Change-Id: I0a88c3a352139fbe2b93f4634794de929826ff47
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8610829
> Reviewed-by: Stephen Chenney <schenney@chromium.org>
> Commit-Queue: Stephen Chenney <schenney@chromium.org>
> Auto-Submit: Fredrik Söderquist <fs@opera.com>
> Cr-Commit-Position: refs/heads/main@{#1722305}

(cherry picked from commit 270d5d328addf47356621fb27f93873d5cd5e96f)

Bug: 575705251,574791952
Change-Id: I0a88c3a352139fbe2b93f4634794de929826ff47
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8615740
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/8078@{#1353}
Cr-Branched-From: c50edc4214d81e4f944c3415d55104a49e46d214-refs/heads/main@{#1706539}

diff --git a/third_party/blink/renderer/core/svg/animation/svg_smil_element.cc b/third_party/blink/renderer/core/svg/animation/svg_smil_element.cc
index f64f6064454f213d4dcb356d9ddd40c155b438b7..fe85e6a427c05dbbfded1a6e7fe71ea3a7c5a85e 100644
--- a/third_party/blink/renderer/core/svg/animation/svg_smil_element.cc
+++ b/third_party/blink/renderer/core/svg/animation/svg_smil_element.cc
@@ -328,7 +328,9 @@ void SVGSMILElement::BuildPendingResource() {
   ClearResourceAndEventBaseReferences();
   DisconnectConditions();
 
-  if (!isConnected()) {
+  // Ignore timed elements that are inserted into a <use> shadow tree. This is
+  // currently possible via a cloned event handler.
+  if (!isConnected() || InUseShadowTree()) {
     // Reset the target element if we are no longer in the document.
     SetTargetElement(nullptr);
     return;
